py3status/modules/velib_metropole.py

```diff
@@ -0,0 +1,293 @@
+                "gpsTopLatitude": max(latitudes),
+                "gpsTopLongitude": max(longitudes),
+                "gpsBotLatitude": min(latitudes),
+                "gpsBotLongitude": min(longitudes),
+            }
+        )
+        self.optimal_area = True
+
+    def _get_velib_data(self):
+        try:
+            return self.py3.request(
+                VELIB_ENDPOINT, params=self.gps_dict, timeout=self.request_timeout
+            ).json()
+        except self.py3.RequestException:
+            return None
+
+    def _manipulate(self, data):
+        new_data = {}
+        stations_data = []
+
+        for code in self.stations:
+            new_station = {}
+
+            # search station
+            station = next(
+                (item for item in data if int(item["station"]["code"]) == int(code)),
+                None,
+            )
+
+            # flat, camel to snake and cast...
+            for key, value in self.py3.flatten_dict(station, delimiter="_").items():
+                key = self._camel_to_snake_case(key)
+                if key in self.placeholders:
+                    new_station[key] = self._cast_number(value)
+
+            # station_due_date_s: station due date in dateime iso format
+            if all("station_due_date" in list for list in [new_station, self.placeholders]):
+                station_due_date = datetime.fromtimestamp(
+                    new_station["station_due_date"]
+                )
+                new_station.update({"station_due_date_s": station_due_date.isoformat()})
+
+            for x in self.thresholds_init['format_station']:
+                if x in new_station:
+                    self.py3.threshold_get_color(new_station[x], x)
+
+            stations_data.append(new_station)
+
+        # forge return
+        for index, station in enumerate(stations_data, 1):
+            new_data[index] = station
+
+        return new_data
+
+    def velib_metropole(self):
+        # refresh
+        current_time = time()
+        refresh = current_time >= self.idle_time
+
+        # time
+        if refresh:
+            self.idle_time = current_time + self.cache_timeout
+            cached_until = self.cache_timeout
+        else:
+            cached_until = self.idle_time - current_time
+
+        if self.scrolling and not refresh:
+            self.scrolling = False
+            data = self.velib_metropole_data
+        else:
+            data = self.velib_metropole_data = self._get_velib_data()
+
+        if data:
+            self.stations_data = self._manipulate(data)
+            if not self.optimal_area:
+                self._set_optimal_area(self.stations_data)
+
+        self.number_of_stations = len(self.stations_data)
+
+        if not self.stations_data:
+            velib_data = {"index": 1, "stations": 0, "format_station": {}}
+        else:
+            # reset station_index counter
+            if self.station_index == 0:
+                self.station_index = 1
```

Comment:
If you have to do this, then your `index` code is probably off. Your index should start at `0`. You should only have to do `+ 1` thing on velib_data's `"index": self.station_index` (and not everywhere)